Makefile.toml

```diff
@@ -0,0 +1,45 @@
+[tasks.server-dev]
```

Comment:
What is the value/use of the `Makefile.toml`?

Reply:
It's explained in the description of the PR. But it's basically a convenient tool to use to automate some repetitive tasks. In this case, this task is to start the server in watch mode in a particular port. So devs don't have to remember or copy paste a long command. Just do `cargo make server-dev`. Or if you want to run the tests in watch mode (in this case for the IDE side of the code), we would just do `cargo make ide-test` or `makers ide-test` (both ways are ok).

One of the nice things about this tool compare to Make files is that it's cross-platform and it's rust based.

Reply:
Can we have this outside of the repo? Seems something nice to have as a developer to test but probably not something we should commit in the repo (and added to `.gitignore`). @jasonforal any thoughts on this?

Reply:
We could have it ignored, that's always an option. But one of the benefits is to provide shared scripts. Another option is to host that file in the IDE folder. That would make things a little less convenient as we would have to use it like this: `makers --makefile crates/bins/src/bin/datadog_static_analyzer_server/ide/Makefile.toml server-dev` instead of just `makers server-dev` but it would work.

Reply:
Let me know what's your preference and I'll move the file if needed.

Reply:
Fair enough. I'll move it to the ide folder as we can then use it there, either by moving the cwd there or by using an alias to not have to type the --makefile path. I thought that it could be convenient for your team to maybe use it for some of the repetitive tasks you have. But I may be biased as I'm very fond of this tool, and use it in almost every project. Not only for complex tasks but even for trivial ones. For instance, in the `udeps` tasks it will automatically change the toolchain to nightly, install the udeps crate for you and run it. I moved it already 😀!